### ISSUE TITLE
Allow chaining variants

### ISSUE THREAD

**Issue by bitspittle** [varabyte/kobweb#121]
So you could do something like

```
Link(variant = UndecoratedLinkVariant.then(UncoloredLinkVariant)) { ... }
```

**Comment by bitspittle** [comment]
This would probably require extracting out a simple interface for ComponentVariant, with `addStyles` and `then`, and then having SingleComponentVariant and CompositeComponentVariant implementations.

**Comment by bitspittle** [comment]
Marked this as good first issue because it's *probably* a relatively straightforward change and could be a good way for a new contributor to stick their hand into one of the more interesting parts of the Kobweb codebase, but I would probably peer program this one, because the surrounding code is a little scary AND maybe this is harder than I'm thinking.

**Comment by bitspittle** [comment]
This was fixed with 3e9223373bcffeca5e17dcb48e1d5fb26b0d52dd
